src/main/java/jenkins/metrics/api/MetricsRootAction.java

```diff
@@ -158,7 +158,7 @@ public HttpResponse doHealthcheck(StaplerRequest req, @QueryParameter("key") Str
             key = getKeyFromAuthorizationHeader(req);
         }
         Metrics.checkAccessKeyHealthCheck(key);
-        return Metrics.cors(key, new HealthCheckResponse(Metrics.healthCheckRegistry().runHealthChecks()));
+        return Metrics.cors(key, new HealthCheckResponse(Metrics.getHealthCheckResults()));
```

Comment:
On the other hand there could be a DoS attack by repeatedly asking for health checks... so we can't just blindly block... some sort of rate limiting or shorter caching time than the periodic work offers might be the solution here